python/ray/serve/scripts.py

```diff
@@ -409,8 +409,25 @@ def run(
             import_attr(import_path), args_dict
         )
 
-    # Setting the runtime_env here will set defaults for the deployments.
-    ray.init(address=address, namespace=SERVE_NAMESPACE, runtime_env=final_runtime_env)
+    # Only initialize ray if it has not happened yet.
+    if not ray.is_initialized():
+        # Setting the runtime_env here will set defaults for the deployments.
+        ray.init(
+            address=address, namespace=SERVE_NAMESPACE, runtime_env=final_runtime_env
+        )
+
+    # Warning users the address they passed is different from the existing ray instance.
+    worker = ray._private.worker.global_worker
+    ray_address = worker.node.address
```

Comment:
for clarity i'd suggest to put this in an `else` block above (this check cannot ever be hit if we call `ray.init` above)